File: hse/BPM161/kaznacheev/2/src/test/java/SingleThreadTest.java
import org.junit.Test;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class SingleThreadTest {

    @Test
    public void isEmptyWorks() {
        LockFreeSetImpl<Integer> set = new LockFreeSetImpl<>();
        set.add(1);
        assertTrue(set.contains(1));
        assertFalse(set.contains(0));
    }

    @Test
    public void removeWorks() {
        LockFreeSetImpl<Integer> set = new LockFreeSetImpl<>();
        set.add(1);
        assertTrue(set.contains(1));
        set.remove(1);
        assertFalse(set.contains(1));
    }

}

File: hse/BPM161/kuporosov/2/src/test/kotlin/net/netau/vasyoid/SingleThreadTest.kt
package net.netau.vasyoid

import org.junit.Assert.*
import org.junit.Test
import java.util.*
import java.util.concurrent.ConcurrentHashMap
import kotlin.random.Random

class SingleThreadTest {

    @Test
    fun stress() {
        val exp = Collections.newSetFromMap(ConcurrentHashMap<Int, Boolean>())
        val act = LockFreeSet<Int>()
        for (i in 0..10000) {
            val value = Random.nextInt(10)
            val action = Random.nextInt(4)
            val (expected, actual) = when (action) {
                0 -> exp.add(value) to act.add(value)
                1 -> exp.remove(value) to act.remove(value)
                2 -> exp.contains(value) to act.contains(value)
                else -> exp.isEmpty() to act.isEmpty()
            }
            assertEquals(expected, actual)
            val expectedElems = exp.iterator().asSequence().toList().sorted()
            val actualElems = act.iterator().asSequence().toList().sorted()
            assertEquals(expectedElems, actualElems)
        }
    }
}